torch_glow/src/PyTorchModelLoader.cpp

```diff
@@ -1598,19 +1655,75 @@ PyTorchModelLoader::PyTorchModelLoader(
     }
   }
 
-  for (const torch::jit::Value *output : graph.outputs()) {
-    glow::NodeValue outputNodeValue;
-    if (auto resOrErr = getGlowNodeValue(output)) {
-      outputNodeValue = std::move(*resOrErr);
+  error = createSaveNodeAndOutputPlaceholders(graph, outputPlaceholders);
+}
+
+/*static*/
+llvm::Error PyTorchModelLoader::parseJITGraph(
+    glow::Function &F, const torch::jit::Graph &graph,
+    const at::ArrayRef<torch::jit::IValue> &inputs,
+    const at::ArrayRef<at::Tensor> &parameters,
+    std::vector<glow::Placeholder *> &inputPlaceholders,
+    std::vector<glow::Placeholder *> &outputPlaceholders) {
+  llvm::Error error = llvm::Error::success();
+  MARK_ERR_CHECKED(error);
+  PyTorchModelLoader loader(F, graph, inputs, parameters, inputPlaceholders,
+                            outputPlaceholders, error);
+  return error;
+}
+
+PyTorchModelLoader::PyTorchModelLoader(
+    glow::Function &F, const torch::jit::Graph &graph,
+    const at::ArrayRef<torch::jit::IValue> &inputs,
+    const at::ArrayRef<at::Tensor> &parameters,
+    std::vector<glow::Placeholder *> &inputPlaceholders,
+    std::vector<glow::Placeholder *> &outputPlaceholders, llvm::Error &error)
+    : F_(F), inputs_(inputs), copyTensorMemory_(true) {
+  auto graphInputValues = graph.inputs();
+  if (inputs.size() + parameters.size() != graphInputValues.size()) {
+    error = MAKE_ERR(glow::strFormat(
+        "Number of Graph inputs %lu must match the "
+        "number of placeholders %lu + number of "
+        "provided inputs %lu.",
+        graphInputValues.size(), parameters.size(), inputs.size()));
+    return;
+  }
+
+  size_t graphIdx = 0;
+  for (size_t i = 0; i < inputs.size(); ++i, ++graphIdx) {
+    const torch::jit::Value *inputValue = graphInputValues[graphIdx];
+    const c10::IValue inputIValue = inputs.at(i);
+    if (inputIValue.isTensor()) {
+      glow::Placeholder *PH;
+      if ((error = createPlaceholderFromTensor(inputValue, PH,
```

Comment:
RETURN_IF_ERR

Reply:
It's a constructor, return error is not allowed.